### PR TITLE
AP_HAL_ChibiOS: add support for Pixracer-periph

### PR DESCRIPTION
**Pull request by ukmook**

### CODE REVIEW

**Comment by magicrub** [comment]
I worked with @ukmook on this and he's tested it thoroughly.. as usual :)

**Comment by magicrub** [comment]
@pkocmoud  This might be of interest to you

**Comment by pkocmoud** [comment]
How did you decide on the board ID? I typically precede this step with a commit to: https://github.com/ArduPilot/ardupilot/blob/master/Tools/AP_Bootloader/board_types.txt

**Comment by magicrub** [comment]
@pkocmoud yes, good call. The ID was chosen by @ukmook , so we'll see what he says, but you're write that it should be added to that list.

**Comment by magicrub** [comment]
@pkocmoud do you have any suggestions or preference on the board ID? Do we have a convention?

**Comment by pkocmoud** [comment]
Typically just the next is sequence.

**Comment by pkocmoud** [comment]
You might want to add this as well. I will confirm there are no other pins that require assignment to prevent increased current draw.

# default to all pins low to avoid ESD issues
DEFAULTGPIO OUTPUT LOW PULLDOWN

**Comment by magicrub** [comment]
It would be nice if we could have this sort of line in the hwdef:
```APJ_BOARD_ID AP_HW_MRO_PIXRACER_PRO_PERIPH```

**Comment by ukmook** [comment]
> @pkocmoud yes, good call. The ID was chosen by @ukmook , so we'll see what he says, but you're write that it should be added to that list.

I chose the next id following on from Cube*_periph, clearly an error on my part

**Comment by magicrub** [comment]
@ukmook not necessarily.

**Comment by magicrub** [comment]
looking at that board ID list, I only see two periph's defined: ```AP_HW_F103_PERIPH``` and ```AP_HW_F303_PERIPH```. I'm tempted to say lets not add it to this PR quite yet because we may want to figure out something special for vehicle boards that are being converted to ap-periph. I'll tag this for a dev call topic but we might get it resolved sooner

**Comment by tridge** [comment]
for boards that normally normally run ardupilot flight control firmware I don't think we want a separate board ID. If we use a separate ID then we can't load the firmware using the normal bootloader as the bootloader refuses firmware without the right board ID. The board ID is about hw compatibility and a periph fw is just like another vehicle type, so doesn't change hw compat

**Comment by magicrub** [comment]
That's true as long as the same bootloader can work for both systems. In the case of 16kb bootloader, can and usb don't fit well together. For example, cube black bootloader needs to be bigger and this requires a different id

**Comment by magicrub** [comment]
@tridge for example, the pixracer DOES NOT define FLASH_RESERVE_START_KB so it uses the default 16. This hwdef uses FLASH_RESERVE_START_KB 64. Therefore, loading AP_Periph app on a standard pixracer will cause the app to not run properly (and vice versa)

**Comment by tridge** [comment]
@magicrub right, if it is a single sector F4 bootloader then we have to have a different ID if we want to use a CAN bootloader

File: libraries/AP_HAL_ChibiOS/hwdef/Pixracer-periph/hwdef-bl.dat [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# hw definition file for processing by chibios_hwdef.py for the
+# mRo Pixracer board as a CANBUS Node AP_Periph.
+
+# MCU class and specific type
+MCU STM32F4xx STM32F427xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1402
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/Pixracer-periph/hwdef-bl.dat:8]: please add patch to board_types.txt

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/Pixracer-periph/hwdef-bl.dat (hunk 1 above)
Comment (magicrub) [reply libraries/AP_HAL_ChibiOS/hwdef/Pixracer-periph/hwdef-bl.dat:8]: added

**Comment by magicrub** [comment]
Added this, and the Cube-periph BoardIDs. Per the dev call, this is ready to merge after it passes CI

**Pull request merged**